packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -977,12 +977,13 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
 
     const render = newProps.children;
 
-    if (typeof render !== 'function') {
-      invariant(
-        false,
-        "A context consumer was rendered with multiple children, or a child that isn't a function. " +
-          'A context consumer expects a single child that is a function. ' +
-          'If you did pass a function, make sure there is no trailing or leading whitespace around it.',
+    if (__DEV__) {
+      warning(
```

Comment:
Should we move this to context creation then? Or do we already warn there?

Reply:
Context creation is too early. Needs to check on every render. Like a prop type.

Reply:
I mean `createElement` when we know type is a context consumer.

Reply:
Ah I see. Idk, not sure what the advantages are. I thought about using `propTypes` but if the plan is to remove those eventually? Although maybe this is an indication that would shouldn't...

Reply:
>Idk, not sure what the advantages are. 

You get a JS stack trace pointing where it happened.